### ISSUE TITLE
valid-jsdoc should recognize typeof

### ISSUE THREAD

**Issue by bdsoha** [eslint/eslint#11130]
**What rule do you want to change?**
> [`valid-jsdoc`](https://eslint.org/docs/rules/valid-jsdoc).

**Does this change cause the rule to produce more or fewer warnings?**
> Fewer.

**How will the change be implemented? (New option, new default behavior, etc.)?**
> Add an option to search for `typeof` within the parameter types.

**Please provide some example code that this change will affect:**
> It fails to recognize `typeof` as a valid type and generates an error for the `valid-jsdoc` rule, for example:
```js
/**
 * @param {typeof SomeClass}
 */
```

**What does the rule currently do for this code?**
> Emits a `valid-jsdoc` warning based on the level set in `.eslintrc`.

**What will the rule do after it's changed?**
> Ignore the `typeof` keyword in parameter types for `jsdoc`.

**Comment by platinumazure** [comment]
Hi @bdsoha, thanks for the issue.

Is this something the JSDoc standard mandates? Or is this something only supported by some JSDoc implementations (e.g., Google Closure Compiler)?

**Comment by nzakas** [comment]
Thanks for the suggestion. As of yesterday, we have decided to officially [end-of-life JSDoc support](https://eslint.org/blog/2018/11/jsdoc-end-of-life) in ESLint. All JSDoc features are now deprecated and we won't be fixing bugs or making any further improvements to those features.

We are recommending that people transition over to use the [`eslint-plugin-jsdoc`](https://github.com/gajus/eslint-plugin-jsdoc) plugin instead of the built-in rules in ESLint.

Thanks for understanding and we apologize for any inconvenience.
